### PR TITLE
WT-9930 Fix checkpoint blocking eviction and causes transactions to all rollback

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

**Comment by quchenhao** [comment]
I haven't been able to reproduce the issue but I believe I found the root cause. If we have a long checkpoint, it will block all the eviction on the tree it is checkpointing. This can take quite a while. If the application thread cannot do eviction in this case, it will return rollback until checkpoint is done on the tree.

**Review by kommiharibabu — APPROVED** [review]
LGTM

**Comment by quchenhao** [comment]
@agorrod Please have a look when you have time. It is not urgent.

File: src/txn/txn.c [hunk 1]
```diff
@@ -2491,6 +2492,14 @@ __wt_txn_is_blocking(WT_SESSION_IMPL *session)
         return (0);
 #endif
 
+    /*
+     * If checkpoint is the oldest id, don't block the transaction as checkpoint may take a long
+     * time to finish. The long checkpoint might also block eviction and cause all the following
+     * transactions to rollback until it finishes.
+     */
+    if (checkpoint_id != WT_TXN_NONE && checkpoint_id == global_oldest)
+        return (0);
```
Comment (agorrod) [inline src/txn/txn.c:2501]: I don't think this fix is precise enough. Generally we need checkpoints to complete, so can't kill them.
It is true that sometimes checkpoint will block eviction in particular btrees, but not in all btrees for the entire duration of the checkpoint.

I'd argue that if a test case (I think this is mostly motivated by Python test failures - is that right?) is triggering this and encountering problems due to the rollback we should reconsider how the Python test is achieving what it wants to do. i.e: overwhelming the cache to such an extent that it becomes `stuck` and looks for user transactions to kill, while a checkpoint is running for long enough to cause a problem. That Python test is probably going to be unreliable and take a long time to run. We should instead find different ways to test the functionality.

**Review by agorrod — CHANGES_REQUESTED** [review]

File: src/txn/txn.c (hunk 1 above)
Comment (quchenhao) [reply src/txn/txn.c:2501]: No, the test failures are format failures. The common theme of the config that failed is that they all have slow checkpoint. They either have the debug option slow_checkpoint on or running the directio test.

**Pull request closed without merging**